compiler/parser/parser.cpp

```diff
@@ -750,8 +757,21 @@ static bool containsOnlyModules(BlockStmt* block, const char* path) {
              lastModSym->name,
              stmtKind);
 
+  } else if (moduleDefs >= 1 && (hasUses || hasOther)) {
+    USR_WARN(firstOtherStmt,
+            "All Chapel code must be contained within a module, "
+            "so an implicit module named '%s' is being introduced "
+            "to store the contents of the file '%s' due to file-scope "
+            "code found on line %i even though the file contains "
+            "the module '%s'",
+            filenameToModulename(path),
```

Comment:
The code itself seems reasonable.  I think this error message should instead say something like:

"All Chapel code must be contained within a module.   Code found on line %i is defined outside of file-scope module '%s', so an implicit module named '%s' is being introduced to contain all the contents of file '%s'"

Reply:
I like Brad's version